front/hooks/useDraftPersistence.ts

```diff
@@ -0,0 +1,181 @@
+import type { JSONContent } from "@tiptap/react";
+import { useCallback, useEffect, useRef, useState } from "react";
+
+const DRAFT_STORAGE_PREFIX = "dust_draft_";
+const DRAFT_EXPIRY_DAYS = 7;
+const STORAGE_DEBOUNCE_MS = 500;
+
+interface DraftData {
+  content: JSONContent;
+  timestamp: number;
+}
+
+interface UseDraftPersistenceOptions {
+  conversationId: string | null;
+  enabled?: boolean;
+}
+
+export function useDraftPersistence({
```

Comment:
`nit`: can we rename to something more explicit like `useDraftMessagePersistence` or something?